### PR TITLE
Implement watermark idea. 

### PR DESCRIPTION
**Pull request by ProjectsByJackHe**
## Description

With [this commit on netperf](https://github.com/microsoft/netperf/commit/883ac249167ad981429e984c5c66e66d54f1d862),

We are now generating a "BestResult" and "Noise" component to our runs.

This PR also adds P0, P50, P99 upper bound checks for latency. 

## Testing

CI

## Documentation

N/A

### CODE REVIEW

File: scripts/secnetperf.ps1 [hunk 1]
```diff
@@ -204,7 +204,7 @@ $allTests["rps-up-512-down-4000"] = "-exec:lowlat -rstream:1 -up:512 -down:4000
 $hasFailures = $false
 $json["run_args"] = $allTests
 
-function CheckRegressionTput($values, $testid, $transport, $regressionJson) {
+function CheckRegressionResult($values, $testid, $transport, $regressionJson) {
```
Comment (nibanks) [inline scripts/secnetperf.ps1:207]: Can we move this function into the helper file? That's where all helper functions are meant to be.

File: scripts/secnetperf.ps1 [hunk 2]
```diff
@@ -215,20 +215,69 @@ function CheckRegressionTput($values, $testid, $transport, $regressionJson) {
...
         Write-Host "No regression baseline found"
         return "NULL"
     }
+
+    try {
+        $BestResult = $regressionJson.$Testid.$envStr.BestResult
+        $Noise = $regressionJson.$Testid.$envStr.Noise
+        $BestResultCommit = $regressionJson.$Testid.$envStr.BestResultCommit
+        if ($avg -lt $baseline) {
+            Write-GHError "🤮 Regression detected in $Testid for $envStr. Baseline: $baseline, New: $avg, BestResult: $BestResult, Noise: $Noise, BestResultCommit: $BestResultCommit"
```
Comment (nibanks) [inline scripts/secnetperf.ps1:230]: For now, let's change all regression detections to warnings, so as to not block progress.

Comment (ProjectsByJackHe) [reply scripts/secnetperf.ps1:230]: This is mainly for printing stuff to console for observability, it doesn't block progress.

the line `run: ./.github/workflows/generate-summary.ps1 --BlockOnFailure ${{ github.event.client_payload.pr != '' && !inputs.commit }}` from https://github.com/microsoft/netperf/blob/main/.github/workflows/quic.yml is what determines if the workflow should stop (exit code 1) or not if there is a regression.

**Comment by ProjectsByJackHe** [comment]
Github seems to be having some technical problems (my latest commits aren't showing up) So gonna try closing this PR and opening a new one.

**Pull request closed without merging**